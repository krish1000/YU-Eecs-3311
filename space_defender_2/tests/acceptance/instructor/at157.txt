--Test Repair is not done if Starfighter is destroyed.
toggle_debug_mode

play(10,10,1,2,2,2,2)
setup_select(3)
setup_next(3)
setup_select(2)
setup_next(1)
setup_next(1)

move(A,1)
pass
special
