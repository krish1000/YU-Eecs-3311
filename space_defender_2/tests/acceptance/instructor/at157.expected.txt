  state:not started, normal, ok
  Welcome to Space Defender Version 2.
->toggle_debug_mode
  state:not started, debug, ok
  In debug mode.
->play(10,10,1,2,2,2,2)
  state:weapon setup, debug, ok
  1:Standard (A single projectile is fired in front)
    Health:10, Energy:10, Regen:0/1, Armour:0, Vision:1, Move:1, Move Cost:1,
    Projectile Damage:70, Projectile Cost:5 (energy)
  2:Spread (Three projectiles are fired in front, two going diagonal)
    Health:0, Energy:60, Regen:0/2, Armour:1, Vision:0, Move:0, Move Cost:2,
    Projectile Damage:50, Projectile Cost:10 (energy)
  3:Snipe (Fast and high damage projectile, but only travels via teleporting)
    Health:0, Energy:100, Regen:0/5, Armour:0, Vision:10, Move:3, Move Cost:0,
    Projectile Damage:1000, Projectile Cost:20 (energy)
  4:Rocket (Two projectiles appear behind to the sides of the Starfighter and accelerates)
    Health:10, Energy:0, Regen:10/0, Armour:2, Vision:2, Move:0, Move Cost:3,
    Projectile Damage:100, Projectile Cost:10 (health)
  5:Splitter (A single mine projectile is placed in front of the Starfighter)
    Health:0, Energy:100, Regen:0/10, Armour:0, Vision:0, Move:0, Move Cost:5,
    Projectile Damage:150, Projectile Cost:70 (energy)
  Weapon Selected:Standard
->setup_select(3)
  state:weapon setup, debug, ok
  1:Standard (A single projectile is fired in front)
    Health:10, Energy:10, Regen:0/1, Armour:0, Vision:1, Move:1, Move Cost:1,
    Projectile Damage:70, Projectile Cost:5 (energy)
  2:Spread (Three projectiles are fired in front, two going diagonal)
    Health:0, Energy:60, Regen:0/2, Armour:1, Vision:0, Move:0, Move Cost:2,
    Projectile Damage:50, Projectile Cost:10 (energy)
  3:Snipe (Fast and high damage projectile, but only travels via teleporting)
    Health:0, Energy:100, Regen:0/5, Armour:0, Vision:10, Move:3, Move Cost:0,
    Projectile Damage:1000, Projectile Cost:20 (energy)
  4:Rocket (Two projectiles appear behind to the sides of the Starfighter and accelerates)
    Health:10, Energy:0, Regen:10/0, Armour:2, Vision:2, Move:0, Move Cost:3,
    Projectile Damage:100, Projectile Cost:10 (health)
  5:Splitter (A single mine projectile is placed in front of the Starfighter)
    Health:0, Energy:100, Regen:0/10, Armour:0, Vision:0, Move:0, Move Cost:5,
    Projectile Damage:150, Projectile Cost:70 (energy)
  Weapon Selected:Snipe
->setup_next(3)
  state:power setup, debug, ok
  1:Recall (50 energy): Teleport back to spawn.
  2:Repair (50 energy): Gain 50 health, can go over max health. Health regen will not be in effect if over cap.
  3:Overcharge (up to 50 health): Gain 2*health spent energy, can go over max energy. Energy regen will not be in effect if over cap.
  4:Deploy Drones (100 energy): Clear all projectiles.
  5:Orbital Strike (100 energy): Deal 100 damage to all enemies, affected by armour.
  Power Selected:Recall (50 energy): Teleport back to spawn.
->setup_select(2)
  state:power setup, debug, ok
  1:Recall (50 energy): Teleport back to spawn.
  2:Repair (50 energy): Gain 50 health, can go over max health. Health regen will not be in effect if over cap.
  3:Overcharge (up to 50 health): Gain 2*health spent energy, can go over max energy. Energy regen will not be in effect if over cap.
  4:Deploy Drones (100 energy): Clear all projectiles.
  5:Orbital Strike (100 energy): Deal 100 damage to all enemies, affected by armour.
  Power Selected:Repair (50 energy): Gain 50 health, can go over max health. Health regen will not be in effect if over cap.
->setup_next(1)
  state:setup summary, debug, ok
  Weapon Selected:Snipe
  Armour Selected:None
  Engine Selected:Standard
  Power Selected:Repair (50 energy): Gain 50 health, can go over max health. Health regen will not be in effect if over cap.
->setup_next(1)
  state:in game(0.0), debug, ok
  Starfighter:
    [0,S]->health:60/60, energy:160/160, Regen:1/7, Armour:1, Vision:22, Move:12, Move Cost:2, location:[E,1]
      Projectile Pattern:Snipe, Projectile Damage:1000, Projectile Cost:20 (energy)
      Power:Repair (50 energy): Gain 50 health, can go over max health. Health regen will not be in effect if over cap.
      score:0
  Enemy:
  Projectile:
  Friendly Projectile Action:
  Enemy Projectile Action:
  Starfighter Action:
  Enemy Action:
  Natural Enemy Spawn:
      1  2  3  4  5  6  7  8  9 10
    A _  _  _  _  _  _  _  _  _  _
    B _  _  _  _  _  _  _  _  _  _
    C _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _
    E S  _  _  _  _  _  _  _  _  _
    F _  _  _  _  _  _  _  _  _  _
    G _  _  _  _  _  _  _  _  _  _
    H _  _  _  _  _  _  _  _  _  _
    I _  _  _  _  _  _  _  _  _  _
    J _  _  _  _  _  _  _  _  _  _
->move(A,1)
  state:in game(1.0), debug, ok
  Starfighter:
    [0,S]->health:60/60, energy:152/160, Regen:1/7, Armour:1, Vision:22, Move:12, Move Cost:2, location:[A,1]
      Projectile Pattern:Snipe, Projectile Damage:1000, Projectile Cost:20 (energy)
      Power:Repair (50 energy): Gain 50 health, can go over max health. Health regen will not be in effect if over cap.
      score:0
  Enemy:
    [1,F]->health:150/150, Regen:5, Armour:10, Vision:10, seen_by_Starfighter:T, can_see_Starfighter:T, location:[A,10]
  Projectile:
  Friendly Projectile Action:
  Enemy Projectile Action:
  Starfighter Action:
    The Starfighter(id:0) moves: [E,1] -> [A,1]
  Enemy Action:
  Natural Enemy Spawn:
    A Fighter(id:1) spawns at location [A,10].
      1  2  3  4  5  6  7  8  9 10
    A S  _  _  _  _  _  _  _  _  F
    B _  _  _  _  _  _  _  _  _  _
    C _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _
    F _  _  _  _  _  _  _  _  _  _
    G _  _  _  _  _  _  _  _  _  _
    H _  _  _  _  _  _  _  _  _  _
    I _  _  _  _  _  _  _  _  _  _
    J _  _  _  _  _  _  _  _  _  _
->pass
  state:in game(2.0), debug, ok
  Starfighter:
    [0,S]->health:60/60, energy:160/160, Regen:1/7, Armour:1, Vision:22, Move:12, Move Cost:2, location:[A,1]
      Projectile Pattern:Snipe, Projectile Damage:1000, Projectile Cost:20 (energy)
      Power:Repair (50 energy): Gain 50 health, can go over max health. Health regen will not be in effect if over cap.
      score:0
  Enemy:
    [1,F]->health:150/150, Regen:5, Armour:10, Vision:10, seen_by_Starfighter:T, can_see_Starfighter:T, location:[A,4]
  Projectile:
    [-1,<]->damage:100, move:10, location:[A,3]
  Friendly Projectile Action:
  Enemy Projectile Action:
  Starfighter Action:
    The Starfighter(id:0) passes at location [A,1], doubling regen rate.
  Enemy Action:
    A Fighter(id:1) moves: [A,10] -> [A,4]
      A enemy projectile(id:-1) spawns at location [A,3].
  Natural Enemy Spawn:
      1  2  3  4  5  6  7  8  9 10
    A S  _  <  F  _  _  _  _  _  _
    B _  _  _  _  _  _  _  _  _  _
    C _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _
    F _  _  _  _  _  _  _  _  _  _
    G _  _  _  _  _  _  _  _  _  _
    H _  _  _  _  _  _  _  _  _  _
    I _  _  _  _  _  _  _  _  _  _
    J _  _  _  _  _  _  _  _  _  _
->special
  state:not started, debug, ok
  Starfighter:
    [0,S]->health:0/60, energy:160/160, Regen:1/7, Armour:1, Vision:22, Move:12, Move Cost:2, location:[A,1]
      Projectile Pattern:Snipe, Projectile Damage:1000, Projectile Cost:20 (energy)
      Power:Repair (50 energy): Gain 50 health, can go over max health. Health regen will not be in effect if over cap.
      score:0
  Enemy:
    [1,F]->health:150/150, Regen:5, Armour:10, Vision:10, seen_by_Starfighter:T, can_see_Starfighter:T, location:[A,4]
  Projectile:
  Friendly Projectile Action:
  Enemy Projectile Action:
    A enemy projectile(id:-1) moves: [A,3] -> [A,1]
      The projectile collides with Starfighter(id:0) at location [A,1], dealing 99 damage.
      The Starfighter at location [A,1] has been destroyed.
  Starfighter Action:
  Enemy Action:
  Natural Enemy Spawn:
      1  2  3  4  5  6  7  8  9 10
    A X  _  _  F  _  _  _  _  _  _
    B _  _  _  _  _  _  _  _  _  _
    C _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _
    F _  _  _  _  _  _  _  _  _  _
    G _  _  _  _  _  _  _  _  _  _
    H _  _  _  _  _  _  _  _  _  _
    I _  _  _  _  _  _  _  _  _  _
    J _  _  _  _  _  _  _  _  _  _
  The game is over. Better luck next time!
